### PR TITLE
Correct int value for faction enum deserialization

### PR DESCRIPTION
**Pull request by Sheikah45**

### CODE REVIEW

**Comment by Sheikah45** [comment]
Review please

**Review by Brutus5000 — COMMENTED** [review]
1) I don't know why we need the string naming double. You can always deserialize from the text value using an object mapper.
2) Again we are still relying on the order of the enums. Why not give the expected server index as a constructor argument instead?

**Comment by Sheikah45** [comment]
can change the index thing. Also not sure what you me by string naming double

**Comment by Brutus5000** [comment]
You have it as a constructor argument and as a @JsonProperty

**Comment by Sheikah45** [comment]
ah yeah missed removing that I had it local but didn't push

**Review by Brutus5000 — APPROVED** [review]

File: faf-commons-lobby/src/main/kotlin/com/faforever/commons/lobby/FafLobbyApi.kt [hunk 1]
```diff
@@ -104,27 +107,40 @@ interface FafLobbyApi :
 /// * SHARED ENUMS *
 /// ****************
 
-enum class Faction {
-  @JsonProperty("uef")
-  UEF,
+enum class Faction(
+  @JsonValue
+  val string: String,
```
Comment (Brutus5000) [inline faf-commons-lobby/src/main/kotlin/com/faforever/commons/lobby/FafLobbyApi.kt]: maybe `faString` for avoiding a semi-reserved word

Comment (Sheikah45) [reply faf-commons-lobby/src/main/kotlin/com/faforever/commons/lobby/FafLobbyApi.kt]: good point

**Pull request merged**